utilities/infra.py

```diff
@@ -809,3 +824,66 @@ def wait_for_isvc_pods(client: DynamicClient, isvc: InferenceService, runtime_na
     """
     LOGGER.info("Waiting for pods to be created")
     return get_pods_by_isvc_label(client=client, isvc=isvc, runtime_name=runtime_name)
+
+
+def assert_dsci_ready(dsci_resource: DSCInitialization) -> None:
+    LOGGER.info(f"Verify DSCI {dsci_resource.name} are {dsci_resource.Status.READY}.")
+    if dsci_resource.status != dsci_resource.Status.READY:
+        raise ResourceNotReadyError(f"DSCI {dsci_resource.name} is not ready.\nStatus: {dsci_resource.instance.status}")
+
+
+def assert_dsc_ready(dsc_resource: DataScienceCluster) -> None:
+    LOGGER.info(f"Verify DSC {dsc_resource.name} are {dsc_resource.Status.READY}.")
+    if dsc_resource.status != dsc_resource.Status.READY:
+        raise ResourceNotReadyError(f"DSCI {dsc_resource.name} is not ready.\nStatus: {dsc_resource.instance.status}")
+
+
+def cluster_sanity(
+    request: FixtureRequest,
+    nodes: list[Node],
+    dsci_resource: DSCInitialization,
+    dsc_resource: DataScienceCluster,
+    junitxml_property: Callable[[str, object], None] | None = None,
+) -> None:
+    """
+    Check cluster resources: nodes, DSCI, DSC and exits pytest execution on failure.
+
+    Args:
+        request (FixtureRequest): pytest request
+        nodes (list[Node]): list of nodes
+        dsci_resource (DSCInitialization): dsci resource
+        dsc_resource (DataScienceCluster): dsc resource
+        junitxml_property (property): Junitxml property
+
+    """
+    skip_cluster_sanity_check = "--cluster-sanity-skip-check"
+    skip_rhoai_check = "--cluster-sanity-skip-rhoai-check"
+
+    if request.session.config.getoption(skip_cluster_sanity_check):
+        LOGGER.warning(f"Skipping cluster sanity check, got {skip_cluster_sanity_check}")
+        return
+
+    try:
+        LOGGER.info("Check cluster sanity.")
+
+        assert_nodes_in_healthy_condition(nodes=nodes, healthy_node_condition_type={"KubeletReady": "True"})
+        assert_nodes_schedulable(nodes=nodes)
+
+        if request.session.config.getoption(skip_rhoai_check):
+            LOGGER.warning(f"Skipping RHOAI resource checks, got {skip_rhoai_check}")
+
+        else:
+            assert_dsci_ready(dsci_resource=dsci_resource)
+            assert_dsc_ready(dsc_resource=dsc_resource)
+
+    except (ResourceNotReadyError, NodeUnschedulableError, NodeNotReadyError) as ex:
+        error_msg = f"Cluster sanity check failed: {str(ex)}"
+        return_code = 99
```

Comment:
Why 99? 😄

Reply:
to be far from https://docs.pytest.org/en/stable/reference/exit-codes.html :)

Reply:
done